merqube_client_lib/templates/equity_baskets/util.py

```diff
@@ -0,0 +1,194 @@
+                    amount=constituent["quantity"],
+                    # this is only at the top level of inline; but its wrong for the cash position (in TP it is CURRENCY_CODE)
+                    identifier_type="CURRENCY_CODE" if sec_type == "CASH" else id_type,
+                )
+            )
+
+        target_port_val = EquityBasketPortfolio(timestamp=date, unit_of_measure=uom, positions=positions)
+
+        EquityBasketPortfolio.parse_obj(target_port_val)  # validate
+        target_portfolios.append((date, target_port_val))
+
+    return target_portfolios
+
+
+def read_file(filename: str) -> list[Any]:
+    """reads portfolio/overrides files"""
+    return list(pd.read_csv(filename, float_precision="round_trip").to_dict(orient="index").values())
+
+
+def log_index(index: IndexDefinitionPost) -> None:
+    """
+    Logs the resulting index
+    """
+    json_formatted_str = json.dumps(json.loads(index.json(exclude_none=True)), indent=4)
+    logger.info("Index spec: \n" + json_formatted_str)
+
+
+def get_index_info(config_file_path: str, required_fields: list[str]) -> dict[str, Any]:
+    """
+    load index specific info from config file + validate the data
+    """
+    index_info = cast(dict[str, Any], json.load(open(config_file_path, "r")))
+
+    for k in required_fields:
+        if k not in index_info:
+            raise ValueError(f"Missing key {k} in index info")
+
+    try:
+        for k in ["base_date"]:
+            pd.Timestamp(index_info[k])
+    except ValueError as e:
+        raise ValueError(
+            "{k} must be a valid date. If the index should run at a certain time of day UTC, that should be part of the ISO8601 formatted first_run_date"
+        ) from e
+
+    if (tz := index_info.get("timezone")) and tz not in pytz.all_timezones:
+        raise ValueError("Invalid timezone string: {tz}")
+
+    return index_info
+
+
+def load_template(
+    template_name: str, config_file_path: str, type_required_specific_fields: list[str]
+) -> tuple[MerqubeAPIClient, IndexDefinitionPost, dict[str, Any], dict[str, Any]]:
+    """
+    Loads a template index model to edit and then create a new index from
+    """
+    index_info = get_index_info(
+        config_file_path=config_file_path, required_fields=type_required_specific_fields + INFO_REQUIRED
+    )
+    token = index_info.get("apikey")
+
+    client = MerqubeAPIClient(token=token)
+    template = client.index_post_model_from_existing(template_name)
+
+    assert template.spec and template.spec.index_class_args and template.spec.index_class_args.get("spec")
+    inner_spec = template.spec.index_class_args["spec"]
+
+    return client, template, index_info, inner_spec
+
+
+def configure_index(template: IndexDefinitionPost, index_info: dict[str, Any], inner_spec: dict[str, Any]) -> None:
+    """
+    set up the index from index info
+    """
+    for k in TOP_LEVEL:
+        setattr(template, k, index_info[k])
```

Comment:
`base_date` format at top level is `%Y/%m/%d`. I'm not exactly sure if this matters. It depends on how the website uses it.

Reply:
done